src/main/java/step4/RacingGameResult.java

```diff
@@ -0,0 +1,39 @@
+package step4;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RacingGameResult {
+
+    private final List<CarStatus> carStatuses;
+
+    private RacingGameResult(List<CarStatus> carStatuses) {
+        this.carStatuses = carStatuses;
+    }
+
+    public static RacingGameResult makeResult(Cars cars) {
+        return new RacingGameResult(cars.getCars().stream()
+                .map(CarStatus::of)
+                .collect(Collectors.toList()));
+    }
+
+    public List<CarStatus> getCarStatuses() {
+        return Collections.unmodifiableList(carStatuses);
+    }
+
+    public List<String> getWinners() {
+        int maxPosition = getMaxPosition();
+        return carStatuses.stream()
+                .filter(carStatus -> maxPosition == carStatus.getPosition())
```

Comment:
```suggestion
                .filter(carStatus -> carStatus.samePosition(maxPosition))
```

상태를 받아서 외부에서 비교하기 보다는 상태를 가진 객체에게 메시지를 전달해보아요 :)